lib/controller.js

```diff
@@ -451,6 +451,45 @@ controller.printAvailableNetworks = function(opts) {
     });
 };
 
+controller.getWifiInfo = function(opts) {
+  return Tessel.get(opts)
+    .then(function(selectedTessel) {
+      return selectedTessel.getWifiInfo()
+        .then(function(network) {
+          // Grab inet lines, flatmap them, remove empty
+          // Wanted to do this with awk and cut inside commands.js
+          var ips = network.ips.filter(function(item) {
+              return /inet/.exec(item);
+            })
+            .map(function(line) {
+              return line.split(' ');
+            })
+            .reduce(function(a, b) {
+              return a.concat(b);
+            })
+            .filter(function(item) {
+              return /addr/.exec(item);
+            })
+            .map(function(chunk) {
+              return chunk.split(':')[1];
+            })
+            .filter(function(addr) {
+              return addr.length;
+            });
+
```

Comment:
I'll change it if that's wanted, but I don't think it's clearer.

On Tue, Sep 8, 2015 at 10:08 AM, Rick Waldron <notifications@github.com>
wrote:

> In lib/controller.js
> <https://github.com/tessel/t2-cli/pull/282#discussion_r38951868>:
>
> > +            .map(function(line) {
> > +              return line.split(' ');
> > +            })
> > +            .reduce(function(a, b) {
> > +              return a.concat(b);
> > +            })
> > +            .filter(function(item) {
> > +              return /addr/.exec(item);
> > +            })
> > +            .map(function(chunk) {
> > +              return chunk.split(':')[1];
> > +            })
> > +            .filter(function(addr) {
> > +              return addr.length;
> > +            });
> > +
>
> :/
>
> This is what reduce is for: consolidating map/filter operations.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tessel/t2-cli/pull/282/files#r38951868>.
>